### PR TITLE
[tesseract] Install additional headers

### PR DESCRIPTION
**Pull request by rozdaybeda**
The main problem is the inability to use the following methods:
- tesseract::Dict::GlobalDawgCache();
- tesseract::SIMDDetect::Update()

for more details see /src/tesseract.cpp, line 692 + line 706:
`...

  // Call GlobalDawgCache here to create the global DawgCache object before
  // the TessBaseAPI object. This fixes the order of destructor calls:
  // first TessBaseAPI must be destructed, DawgCache must be the last object.
  tesseract::Dict::GlobalDawgCache();

...

  // SIMD settings might be overridden by config variable.
  tesseract::SIMDDetect::Update();
...
`

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/tesseract/portfile.cmake [hunk 1]
```diff
@@ -65,5 +65,42 @@ endif()
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
+# Install:
+file(GLOB API_HEADER_FILES LIST_DIRECTORIES false "${SOURCE_PATH}/src/api/*.h")
+file(INSTALL ${API_HEADER_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/include/tesseract/api)
```
Comment (Cheney-W) [inline ports/tesseract/portfile.cmake:70]:
```suggestion
file(INSTALL ${API_HEADER_FILES} DESTINATION "${CURRENT_PACKAGES_DIR}/include/tesseract/api")
```

Comment (Cheney-W) [reply ports/tesseract/portfile.cmake:70]: Please add double quotes to the relative path after `DESTINATION`. 
The following `file(INSTALL xxx DESTINATION xxx)` also needs the same operation.

File: ports/tesseract/portfile.cmake [hunk 2]
```diff
@@ -65,5 +65,42 @@ endif()
...
+file(INSTALL ${CCUTIL_HEADER_FILES} DESTINATION "${CURRENT_PACKAGES_DIR}/include/tesseract/ccutil")
+
+file(GLOB CLASSIFY_HEADER_FILES LIST_DIRECTORIES false "${SOURCE_PATH}/src/classify/*.h")
+file(INSTALL ${CLASSIFY_HEADER_FILES} DESTINATION "${CURRENT_PACKAGES_DIR}/include/tesseract/classify")
+
+file(GLOB CUTIL_HEADER_FILES LIST_DIRECTORIES false "${SOURCE_PATH}/src/cutil/*.h")
+file(INSTALL ${CUTIL_HEADER_FILES} DESTINATION "${CURRENT_PACKAGES_DIR}/include/tesseract/cutil")
+
+file(GLOB DICT_HEADER_FILES LIST_DIRECTORIES false "${SOURCE_PATH}/src/dict/*.h")
+file(INSTALL ${DICT_HEADER_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/include/tesseract/dict)
```
Comment (Cheney-W) [inline ports/tesseract/portfile.cmake:91]:
```suggestion
file(INSTALL ${DICT_HEADER_FILES} DESTINATION "${CURRENT_PACKAGES_DIR}/include/tesseract/dict")
```

File: ports/tesseract/portfile.cmake (hunk 1 above)
Comment (rozdaybeda) [reply ports/tesseract/portfile.cmake]: Good point, i fixed it, thanks

File: ports/tesseract/portfile.cmake (hunk 2 above)
Comment (Cheney-W) [reply ports/tesseract/portfile.cmake:91]: @rozdaybeda  Could you please address this review suggestion? Thanks!

**Comment by Cheney-W** [comment]
Pinging @rozdaybeda  Could you please address the review suggestion? Thanks!

**Comment by Cheney-W** [comment]
There is a CI baseline issue, we are fixing it.

**Review by Cheney-W — APPROVED** [review]

File: ports/tesseract/portfile.cmake [hunk 3]
```diff
@@ -65,5 +65,42 @@ endif()
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
+# Install:
+file(GLOB API_HEADER_FILES LIST_DIRECTORIES false "${SOURCE_PATH}/src/api/*.h")
```
Comment (BillyONeal) [inline ports/tesseract/portfile.cmake:69]: I would have preferred to see this loop with `foreach(... IN ITEMS api arch ccmain ccstruct)` etc. but this already passed CI so I'm going to merge it :)

**Review by BillyONeal — APPROVED** [review]
Thanks for your contribution to vcpkg!

**Pull request merged**